sumologic/sumologic.py

```diff
@@ -20,53 +21,59 @@ def __init__(self, accessId, accessKey, endpoint=None, cookieFile='cookies.txt')
             self.endpoint = self._get_endpoint()
         else:
             self.endpoint = endpoint
-        if endpoint[-1:] == "/":
-          raise Exception("Endpoint should not end with a slash character")
+        if self.endpoint[-1:] == "/":
+            raise Exception("Endpoint should not end with a slash character")
 
     def _get_endpoint(self):
         """
         SumoLogic REST API endpoint changes based on the geo location of the client.
         For example, If the client geolocation is Australia then the REST end point is
         https://api.au.sumologic.com/api/v1
-
         When the default REST endpoint (https://api.sumologic.com/api/v1) is used the server
         responds with a 401 and causes the SumoLogic class instantiation to fail and this very
         unhelpful message is shown 'Full authentication is required to access this resource'
-
         This method makes a request to the default REST endpoint and resolves the 401 to learn
         the right endpoint
         """
-        self.endpoint = 'https://api.sumologic.com/api/v1'
+        self.endpoint = 'https://api.sumologic.com/api'
         self.response = self.session.get('https://api.sumologic.com/api/v1/collectors')  # Dummy call to get endpoint
-        endpoint = self.response.url.replace('/collectors', '')  # dirty hack to sanitise URI and retain domain
+        endpoint = self.response.url.replace('/v1/collectors', '')  # dirty hack to sanitise URI and retain domain
+        print("SDK Endpoint", endpoint)
         return endpoint
 
-    def delete(self, method, params=None):
-        r = self.session.delete(self.endpoint + method, params=params)
+    def get_versioned_endpoint(self, version):
+        return self.endpoint+'/%s' % version
+
+    def delete(self, method, params=None, version=DEFAULT_VERSION):
```

Comment:
yes we can do that but do you also want to expose this to users for initialization because I wanted to refrain it from exposing this to users so that they do not have worry about API version

Reply:
Yes, shouldn't matter if we have a default value. If we consider different API versions on Sumo side then clients DO have to worry about the version anyway.